Assets/MixedRealityToolkit/_Core/Devices/BaseController.cs

```diff
@@ -23,14 +23,39 @@ public abstract class BaseController : IMixedRealityController
         /// <param name="controllerHandedness"></param>
         /// <param name="inputSource"></param>
         /// <param name="interactions"></param>
-        protected BaseController(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+        protected BaseController(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource, MixedRealityInteractionMapping[] interactions)
         {
             TrackingState = trackingState;
             ControllerHandedness = controllerHandedness;
             InputSource = inputSource;
             Interactions = interactions;
+
+            // TODO: consider if these should be passed in or not
```

Comment:
I'm inclined to say no, only bc we'd be updating our controller right after we create it anyway.

Reply:
I was thinking some controllers might not use the inputSource, and would want to hard-code their values. Now that I think about it, those controllers can set the desired values in THEIR constructor after calling the base.